File: javabots/SimpleBot/src/anbf/javabots/SimpleBot.java
package anbf.javabots;

import anbf.bot.*;

/** A dumb example bot.  Equivalent to simplebot.clj */
public class SimpleBot {
	IANBF anbf;
	
	class CircleMover implements IActionHandler {
		Direction[] smallCircle = {Direction.N, Direction.E, Direction.S, Direction.W};
		int next = 0;
		
		@Override
		public IAction chooseAction(IGame game) {
			next = next % smallCircle.length;
			//return Actions.Contribute(smallCircle[next++]);
			return Actions.Move(smallCircle[next++]);
		}
	}
	
	class PrayForFood implements IActionHandler {
		@Override
		public IAction chooseAction(IGame game) {
			if (game.player().hunger() == Hunger.FAINTING)
				return Actions.Pray();
			else
				return null;
		}
	}
	
	public SimpleBot(IANBF anbf) {
		this.anbf = anbf;
		anbf.registerHandler(new IChooseCharacterHandler() {
			@Override
			public String chooseCharacter() {
				return "nsm"; // choose samurai
			}
		});
		anbf.registerHandler(0, new PrayForFood());
		anbf.registerHandler(1, new CircleMover());
		System.out.println("bot initialized");
	}
}

File: src/anbf/bots/simplebot.clj
(ns anbf.bots.simplebot
  "A dumb example bot.  Equivalent to SimpleBot.java"
  (:require [clojure.tools.logging :as log]
            [anbf.anbf :refer :all]
            [anbf.delegator :refer :all]
            [anbf.actions :refer :all]))

(def ^:private circle-small (cycle [:N :E :S :W]))

(defn- circle-mover []
  (let [circle (atom circle-small)]
    (reify ActionHandler
      (choose-action [_ _]
        (->Move (first (swap! circle rest)))))))

(defn- pray-for-food []
  (reify ActionHandler
    (choose-action [_ game]
      (if (fainting? (:player game))
        (->Pray)))))

(defn init [anbf]
  (-> anbf
      (register-handler (reify ChooseCharacterHandler
                          (choose-character [this]
                            (deregister-handler anbf this)
                            "nsm"))) ; choose samurai
      (register-handler 0 (pray-for-food))
      (register-handler 1 (circle-mover))))
